src/components/SearchView.tsx

```diff
@@ -0,0 +1,66 @@
+import * as React from "react";
+import { LibraryItem } from "./LibraryItem";
+import { LibraryView } from "../LibraryView";
+import { searchItemResursive, resetItemData, ItemData } from "../LibraryUtilities";
+
+interface SearchModeChangedFunc {
+    (inSearchMode: boolean): void;
+}
+
+interface SearchViewProps {
+    onSearchModeChanged: SearchModeChangedFunc;
+    libraryView: LibraryView;
+    items: ItemData[];
+}
+
+interface SearchViewStates {
+    searchText: string;
+}
+
+export class SearchView extends React.Component<SearchViewProps, SearchViewStates> {
+
+    constructor(props: SearchViewProps) {
+        super(props);
+
+        this.state = {
+            searchText: ''
+        };
+    }
+
+    generateStructuredItems() {
+        let structuredItems: JSX.Element[] = [];
+        let index = 0;
+        return this.props.items.map((item: ItemData) =>
+            <LibraryItem key={index++} libraryView={this.props.libraryView} data={item} />);
+    }
+
+    onTextChange(event: any) {
+        let text = event.target.value.trim().toLowerCase();
+        let hasText = text.length > 0;
+
+        this.setState({ searchText: text });
+
+        // Update LibraryContainer of the search
+        this.props.onSearchModeChanged(hasText);
+    }
+
+    render() {
+        let listItems: JSX.Element[] = [];
+        if (this.state.searchText.length > 0) {
+            searchItemResursive(this.props.items, this.state.searchText);
+            listItems = this.generateStructuredItems();
+        } else {
+            // Reset ItemData when search text is cleared
+            resetItemData(this.props.items);
+        }
+
+        return (
+            <div className="searchView">
+                <div className="searchBar">
+                    <input id="searchInput" type="search" placeholder="Search..." onChange={this.onTextChange.bind(this)}></input>
```

Comment:
Hi @fanwgwg, can you make a task for this? We want `onChange` to not trigger a search so quickly, it should start a timer of maybe `300ms` and if that time expired then we will trigger the search. Otherwise if a new character is being pressed, then the timer will restart to be `300ms`. Thanks!